runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java

```diff
@@ -222,4 +329,63 @@ private static String removeTags(String str)
 		matcher.appendTail(stringBuffer);
 		return stringBuffer.toString();
 	}
+
+	private void timeoutEmojiMessages(boolean force)
+	{
+		if (currentMessage == null)
+		{
+			return;
+		}
+
+		boolean removed = false;
+		MessageNode messageNode = currentMessage.getMessageNode();
+		final int createdTick = currentMessage.getTick();
+
+		if (client.getTickCount() > createdTick + (config.emojiEmitDuration() * 1000 / GAME_TICK_LENGTH) || force)
+		{
+			ChatLineBuffer ccInfoBuffer = client.getChatLineMap().get(ChatMessageType.GAMEMESSAGE.getType());
+			if (ccInfoBuffer != null)
+			{
+				ccInfoBuffer.removeMessageNode(messageNode);
+				removed = true;
+				currentMessage = null;
+			}
+		}
+
+		// don't force a redraw if we're about to add the message back
+		if (removed && !force)
+		{
+			clientThread.invoke(() -> client.runScript(ScriptID.BUILD_CHATBOX));
+		}
+	}
+
+	private String getEmojiMessage()
+	{
+		ChatMessageBuilder builder = new ChatMessageBuilder();
+		ChatMessageBuilder spacer = new ChatMessageBuilder();
```

Comment:
Move this inside the loop and remove the `spacer = new ChatMessageBuilder();` line at the end of the loop